sycl/include/sycl/accessor.hpp

```diff
@@ -2039,20 +2039,17 @@ class __SYCL_EBO __SYCL_SPECIAL_CLASS __SYCL_TYPE(accessor) accessor :
   bool empty() const noexcept { return size() == 0; }
 
   template <int Dims = Dimensions, typename = std::enable_if_t<(Dims > 0)>>
```

Comment:
I think it would still be nice to add 
```suggestion
  template <int Dims = Dimensions, typename = std::enable_if_t<Dims == Dimensions && (Dims > 0)>>
```
and in `get_offset` as well. Should help us avoid misuse in the future.